core-bundle/src/EventListener/CsrfTokenCookieSubscriber.php

```diff
@@ -115,7 +115,7 @@ private function requiresCsrf(Request $request, Response $response): bool
             return true;
         }
 
-        if ($request->hasSession() && $request->getSession()->isStarted()) {
+        if ($request->hasSession() && $request->getSession()->isStarted() && [] !== $request->getSession()->all()) {
```

Comment:
`->all()` only seems to refer to the data in the attributes bag. But we should check all session bags here (like the flash bag).

```suggestion
        if ($request->hasSession() && $request->getSession()->isStarted() && !$request->getSession()->isEmpty()) {
```

`isEmpty()` is marked as internal, but I don’t know how to check that otherwise as there seems to be no way to iterate over all registered bags.

Reply:
Hmm...yes, I've updated the PR. Not the most beautiful but the only way there is as far as I can see.

Reply:
Symfony's `AbstractSessionListener` uses the following check in its `kernel.response` listener:

```php
$isSessionEmpty = ($session instanceof Session ? $session->isEmpty() : !$session->all()) && empty($_SESSION); // checking $_SESSION to keep compatibility with native sessions
```

Do we need to employ the same fallbacks? 🤔

Reply:
I don't think in our case we want to support the `$_SESSION` case. We have deprecated its usage long ago and removed all the ArrayAccess stuff in v5 so it wouldn't really work with our CSRF framework anyway.